pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/parser/RuntimeASTTransformer.groovy

```diff
@@ -606,7 +610,8 @@ class RuntimeASTTransformer {
                     transformOptions(original.options),
                     transformTriggers(original.triggers),
                     transformParameters(original.parameters),
-                    transformLibraries(original.libraries)))
+                    transformLibraries(original.libraries),
+                    constX(original.stages.hashCode())))
```

Comment:
Humor me and use CRC32 if not SHA-*. `String.hashCode` is pretty weak. Probably this is not a security consideration but are you willing to stake your reputation on that?

Reply:
So the only reason I'm actually using `hashCode` is that it's trivial to get off an object. If there's a better method I should be calling, point me to it and I will *happily* use it.

Reply:
And also, yes, I actually am willing to stake my reputation that this is not a security consideration - the hash code is just used to try to determine which of the possibly multiple `ModelASTStages` instances in `ExecutionModelAction` is the one we actually invoked, so that we can make `ExecutionModelAction.getStages()` return that instance. And that method is only used by Blue Ocean to pre-render the stages that *will* run before they've actually run. So worst case is that the wrong group of stages renders initially.

That said, I'd be fine with literally anything as a unique identifier for a `ModelASTStages` instance - if there's a good way to randomly generate a string or whatever at instantiation time that I can add as a field to `ModelASTStages`, that'd be perfect. I just went with `hashCode()` because I knew it'd work...and I knew you'd eventually tell me the right way I should do it. =)

Reply:
Switched around to store a `UUID` on `ModelASTStages` that we generate at initialization, and a `String` representation of it on `Root` (to make the sandboxed transformation stuff easier than trying to instantiate a `UUID` from the existing `ModelASTStages.getUuid()` `String`).